aws/src/main/java/org/apache/iceberg/aws/AwsClientFactories.java

```diff
@@ -118,6 +122,22 @@ public void initialize(Map<String, String> properties) {
       ValidationException.check((s3AccessKeyId == null && s3SecretAccessKey == null) ||
           (s3AccessKeyId != null && s3SecretAccessKey != null),
           "S3 client access key ID and secret access key must be set at the same time");
+      this.clientHttpType = PropertyUtil.propertyAsString(properties,
+          AwsProperties.CLIENT_HTTP_TYPE, AwsProperties.CLIENT_HTTP_TYPE_DEFAULT);
+    }
+  }
+
+  static SdkHttpClient.Builder configureHttpClientBuilder(String clientHttpType) {
+    if (Strings.isNullOrEmpty(clientHttpType)) {
+      return  UrlConnectionHttpClient.builder();
```

Comment:
should we return what is present in default `AwsProperties.CLIENT_HTTP_TYPE_DEFAULT` here rather than Url client always ?

Reply:
good point! fixed. thanks.